### PR TITLE
[sdk][build] expand board support and update to GSDK 4.1.0

### PR DESCRIPTION
**Pull request by lmnotran**
This expands board support to a larger selection through the use of the Silabs tool [`slc_cli`](https://www.silabs.com/documents/public/user-guides/ug520-software-project-generation-configuration-with-slc-cli.pdf).

### Change summary
- Add `slc_cli` to bootstrap and build scripts
- Templatize platform and mbedtls CMake libraries using `slc_cli` and jinja template
- Update to Gecko SDK 4.1.0
  - Also switch from [`sdk_support`](https://github.com/SiliconLabs/sdk_support) (a lite release of GSDK) to [`gecko_sdk`](https://github.com/SiliconLabs/gecko_sdk) (a full GSDK release).
  - `git-lfs` is now required 
- Change build command format. [Platform is now parsed from the board's `.slcc` file.](https://github.com/SiliconLabs/ot-efr32/pull/438/files#diff-99b83631df533ea1d9c97396a043a4670464f523d0c1e60b0a1cab6dddd440f9R224-R238) 
  - OLD: `./script/build efr32mgXX -DBOARD=brdXXXXy`
  - NEW: `./script/build brdXXXXy`
- [Change build output folder](https://github.com/SiliconLabs/ot-efr32/pull/438/files#diff-99b83631df533ea1d9c97396a043a4670464f523d0c1e60b0a1cab6dddd440f9R69)
  - OLD: `ot-efr32/build/efr32mgXX`
  - NEW: `ot-efr32/build/brdXXXXy`
- Add `brd4186c` and `brd4187c` to [`./script/test`](https://github.com/SiliconLabs/ot-efr32/pull/438/files#diff-2ade384ce3c8dda6cc27f888849ab9af75d1870698b5d6e5b7d2e5d1ba86d064R73-R74)
- Add [`Dockerfile`](https://github.com/SiliconLabs/ot-efr32/pull/438/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557). The docker image may be used by jenkins. It could also be used to define a devcontainer.json later to provide a development ready environment.

### Previous incremental PRs
- https://github.com/SiliconLabs/ot-efr32/pull/425
- https://github.com/SiliconLabs/ot-efr32/pull/426
- https://github.com/SiliconLabs/ot-efr32/pull/433
- https://github.com/SiliconLabs/ot-efr32/pull/438

### Known problems
- This breaks the `sleepy-demo` example application build. A fix PR will be following this one shortly

### CODE REVIEW

**Review by romacdon — APPROVED** [review]
LGTM

**Comment by hemanth-silabs** [comment]
LGTM 👍

**Review by hemanth-silabs — APPROVED** [review]

**Comment by lmnotran** [comment]
@jwhui can you take a quick look at this PR? There's one last thing I need to check before we merge, but I figured I would ping you so you could get started with the review since it's so large.

**Comment by lmnotran** [comment]
@jwhui Just finished my final checks. Feel free to merge whenever you're ready!

File: script/slc_cli [hunk 1]
```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+#
+#  Copyright (c) 2021, The OpenThread Authors.
```
Comment (jwhui) [inline script/slc_cli:3]: nit:
```suggestion
#  Copyright (c) 2022, The OpenThread Authors.
```

**Pull request merged**